src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+            is SuperstructureRequest.GroundIntakeCone ->
+              SuperstructureStates.GROUND_INTAKE_CONE_PREP
+            is SuperstructureRequest.GroundIntakeCube ->
+              SuperstructureStates.GROUND_INTAKE_CUBE_PREP
+            is SuperstructureRequest.Home -> SuperstructureStates.HOME_PREP
+            is SuperstructureRequest.Idle -> SuperstructureStates.IDLE
+            is SuperstructureRequest.PrepScore -> {
+              when (usingGamePiece) {
+                GamePiece.CONE -> SuperstructureStates.SCORE_PREP
+                GamePiece.CUBE -> SuperstructureStates.SCORE_PREP
+                else -> {
+                  currentRequest = SuperstructureRequest.Idle()
+                  SuperstructureStates.IDLE
+                }
+              }
+            }
+            is SuperstructureRequest.SingleSubstationIntake -> {
+              when (usingGamePiece) {
+                GamePiece.CONE -> SuperstructureStates.SINGLE_SUBSTATION_INTAKE_CONE
+                GamePiece.CUBE -> SuperstructureStates.SINGLE_SUBSTATION_INTAKE_CUBE
+                GamePiece.NONE -> SuperstructureStates.IDLE
+              }
+            }
+            is SuperstructureRequest.Tuning -> SuperstructureStates.TUNING
+            else -> currentState
+          }
+      }
+      SuperstructureStates.HOME_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedDownAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition) {
+          nextState = SuperstructureStates.HOME
+        }
+      }
+      SuperstructureStates.HOME -> {
+        // Outputs
+        elevator.currentRequest = Request.ElevatorRequest.Home()
+        manipulator.currentRequest = Request.ManipulatorRequest.Home()
+
+        // Transition
+        if (elevator.isHomed && manipulator.isHomed) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.intakeAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.groundIntakeCubeExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
```

Comment:
we should check if the request is not `GroundIntakeCube`, not just if it is `Idle` (imagine if you hit the ground intake button, then press the score button or something without lifting, idk)

Reply:
to be clear -- we should still transition to `CLEANUP` regardless, just that we should clean up if we are requesting anything other than `GroundIntakeCube`